core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -184,6 +185,46 @@ public MysqlSqlDialect(Context context) {
     return super.getCastSpec(type);
   }
 
+  /** {@inheritDoc}
+   *
+   * <p>MySQL format element reference:
+   * <a href="https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format">
+   * MySQL Date and Time Functions</a>.
+   */
+  @Override public String getFormatElement(FormatElementEnum fmtElement) {
```

Comment:
Why does this logic go in MySqlSqlDialect?

Reply:
Each dialect might have a different set of elements to represent the same type of datetime unit. 

For example an abbreviated month name like `Jan` is `%b` in MySQL and `MON` in Snowflake. Each `SqlDialect` will have to override this method to supply the correct string for the element type. I only did MySQL, BigQuery, and Snowflake in the first commit because I wanted to get some feedback before working on each dialect.

Oracle's format elements were chosen as the "base" set since they map closely to the SQL:2016 standard as described in [CALCITE-2980](https://issues.apache.org/jira/browse/CALCITE-2980).

Reply:
Makes sense! Looks great.

Reply:
As I said previously, the association should be between the FORMAT function (as specified by MySQL) and the format element, not intermediated by the dialect object.